WordPress/Classes/ViewRelated/Blaze Campaigns/BlazeCampaignsViewController.swift

```diff
@@ -58,17 +51,111 @@ final class BlazeCampaignsViewController: UIViewController, NoResultsViewHost {
 
     override func viewDidLoad() {
         super.viewDidLoad()
+
         setupView()
         setupNavBar()
         setupNoResults()
+
+        stream.delegate = self
+        stream.load()
+
+        // Refresh data automatically when new campaign is created
+        NotificationCenter.default.addObserver(self, selector: #selector(setNeedsToRefreshCampaigns), name: .blazeCampaignCreated, object: nil)
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        tableView.sizeToFitFooterView()
+    }
+
+    // MARK: - Stream
+
+    func stream(_ stream: BlazeCampaignsStream, didAppendItemsAt indexPaths: [IndexPath]) {
+        // Make sure the existing cells are not reloaded to avoid interfering with image loading
+        UIView.performWithoutAnimation {
+            tableView.insertRows(at: indexPaths, with: .none)
+        }
+    }
+
+    func streamDidRefreshState(_ stream: BlazeCampaignsStream) {
+        reloadView()
+    }
+
+    private func reloadView() {
+        reloadStateView()
+        reloadFooterView()
+        tableView.sizeToFitFooterView()
+    }
+
+    private func reloadStateView() {
+        hideNoResults()
+        noResultsViewController.hideImageView(true)
+        if stream.campaigns.isEmpty {
+            if stream.isLoading {
+                noResultsViewController.hideImageView(false)
+                showLoadingView()
+            } else if stream.error != nil {
+                showErrorView()
+            } else {
+                showNoResultsView()
+            }
+        }
+    }
+
+    private func reloadFooterView() {
+        guard !stream.campaigns.isEmpty else {
+            tableView.tableFooterView = nil
+            return
+        }
+        if stream.isLoading {
+            tableView.tableFooterView = PagingFooterView(state: .loading)
+        } else if stream.error != nil {
+            let footerView = PagingFooterView(state: .error)
+            footerView.buttonRetry.addTarget(self, action: #selector(buttonRetryTapped), for: .touchUpInside)
+            tableView.tableFooterView = footerView
+        } else {
+            tableView.tableFooterView = nil
+        }
+    }
+
+    // MARK: - Actions
+
+    @objc private func buttonRetryTapped() {
+        stream.load()
+    }
+
+    @objc private func setNeedsToRefreshCampaigns() {
+        guard pendingStream == nil else { return }
```

Comment:
This line covers the following (rare?) scenario:

- Create a campaign and return to the campaign list; the list starts silently refreshing in the background
- Manually perform PTR

Expected result:
- Only one request is made
- PTR is hidden when is finishes